web/profiles/custom/yalesites_profile/modules/custom/ys_node_access/src/EventSubscriber/NodeAccessEventSubscriber.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Drupal\ys_node_access\EventSubscriber;
+
+use Drupal\Core\EventSubscriber\HttpExceptionSubscriberBase;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Drupal\path_alias\AliasManager;
+use Drupal\Core\Entity\EntityTypeManager;
+
+/**
+ * Overrides the 403 response to perform a CAS redirect for specific pages.
+ */
+class NodeAccessEventSubscriber extends HttpExceptionSubscriberBase {
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The path alias manager.
+   *
+   * @var \Drupal\path_alias\AliasManager
+   */
+  protected $pathAlias;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The construct for this class.
+   *
+   * @param Drupal\Core\Session\AccountInterface $current_user
+   *   The current user service.
+   * @param Drupal\path_alias\AliasManager $path_alias
+   *   The path alias manager.
+   * @param Drupal\Core\Entity\EntityTypeManager $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(AccountInterface $current_user, AliasManager $path_alias, EntityTypeManager $entity_type_manager) {
+    $this->currentUser = $current_user;
+    $this->pathAlias = $path_alias;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * Request format that will be handled by this subscriber.
+   */
+  protected function getHandledFormats() {
+    return ['html'];
+  }
+
+  /**
+   * Manage the 403 exceptions.
+   *
+   * @param Symfony\Component\HttpKernel\Event\ExceptionEvent $event
+   *   The Incoming event.
+   */
+  public function on403(ExceptionEvent $event) {
+    if ($this->currentUser->isAnonymous()) {
+      // Get path alias.
+      $path = $event->getRequest()->getPathInfo();
+
+      // Convert path alias into node object.
+      $internalPath = $this->pathAlias->getPathByAlias($path);
+      $params = Url::fromUri("internal:" . $internalPath)->getRouteParameters();
+      $entityType = key($params);
+
+      // Make sure this is a node.
+      if ($entityType == 'node') {
+        /** @var \Drupal\node\Entity $node */
+        $node = $this->entityTypeManager->getStorage($entityType)->load($params[$entityType]);
+
+        /* Check to see if node is set to require login only.
+         * If so, redirect to CAS login.
+         */
+        if ($node->hasField('field_login_required')) {
+          if ($node->get('field_login_required')->getValue()[0]['value']) {
```

Comment:
Maybe here you could use:` $node->field_login_required->isEmpty() ` to validate if there is data into the field

Reply:
Great idea - since this is only a boolean, we just need to check if it's empty or not. Refactored.